pymc_marketing/mmm/components/base.py

```diff
@@ -0,0 +1,261 @@
+            for parameter_name, variable_name in self.variable_mapping.items()
+        }
+
+    def _checks(self) -> None:
+        self._has_all_attributes()
+        self._function_works_on_instances()
+        self._has_defaults_for_all_arguments()
+
+    def _has_all_attributes(self) -> None:
+        if not hasattr(self, "prefix"):
+            raise NotImplementedError("prefix must be implemented in the subclass")
+
+        if not hasattr(self, "default_priors"):
+            raise NotImplementedError(
+                "default_priors must be implemented in the subclass"
+            )
+
+        if not hasattr(self, "function"):
+            raise NotImplementedError("function must be implemented in the subclass")
+
+    def _has_defaults_for_all_arguments(self) -> None:
+        function_signature = signature(self.function)
+
+        # Remove the first one as assumed to be the data
+        parameters_that_need_priors = set(
+            list(function_signature.parameters.keys())[1:]
+        )
+        parameters_with_priors = set(self.default_priors.keys())
+
+        missing_priors = parameters_that_need_priors - parameters_with_priors
+        missing_parameters = parameters_with_priors - parameters_that_need_priors
+
+        if missing_priors or missing_parameters:
+            raise ParameterPriorException(missing_priors, missing_parameters)
+
+    def _function_works_on_instances(self) -> None:
+        class_function = self.__class__.function
+        function_parameters = list(signature(class_function).parameters)
+
+        is_method = function_parameters[0] == "self"
+        data_parameter_idx = 1 if is_method else 0
+
+        has_data_parameter = (
+            function_parameters[data_parameter_idx] in RESERVED_DATA_PARAMETER_NAMES
+        )
+        if not has_data_parameter:
+            raise MissingDataParameter()
+
+        if is_method:
+            return
+
+        self.function = class_function
+
+    @property
+    def variable_mapping(self) -> dict[str, str]:
+        """Mapping from parameter name to variable name in the model."""
+        return {
+            parameter: f"{self.prefix}_{parameter}"
+            for parameter in self.default_priors.keys()
+        }
+
+    def _create_distributions(self, dim_name: str) -> dict[str, pt.TensorVariable]:
+        distributions: dict[str, pt.TensorVariable] = {}
+        for parameter_name, variable_name in self.variable_mapping.items():
+            parameter_prior = self.function_priors[parameter_name]
+
+            distribution = _get_distribution_from_dict(
+                dist=parameter_prior,
+            )
+
+            distributions[parameter_name] = distribution(
+                name=variable_name,
+                dims=dim_name,
+                **parameter_prior["kwargs"],
+            )
+
+        return distributions
+
+    def apply(self, x: pt.TensorLike, dim_name: str = "channel") -> pt.TensorVariable:
```

Comment:
Thought here: make this dim and actually use a default of None. 
Then just hard-code "channel" only in the Model itself. 

Having default of None can make for easy use independently. ie.
```python
saturation = MichaelisMenten()
x = np.linspace(0, 10)
with pm.Model(): 
    y = saturation.apply(x=x) # currently have to use dim_name=None
```

I have thought of adding plotting methods for both `AdstockTransformation` and `SaturationTransformation`
```python
saturation = MichaelisMenten()
prior_parameters = saturation.sample_prior()
saturation.plot_curve(prior_parameters)
```
Could also be used with `posterior` instead of a prior

Reply:
Also like dim over dim_name since that is the PyMC convention